src/Auth0Client.ts

```diff
@@ -953,6 +988,12 @@ export default class Auth0Client {
 
       const decodedToken = this._verifyIdToken(tokenResult.id_token, nonceIn);
 
+      if (!decodedToken.claims.org_id) {
+        this.cookieStorage.remove(
+          buildOrganizationHintCookieName(this.options.client_id)
+        );
+      }
```

Comment:
Why are we not setting the cookie here? I assume it is because we know it is already set at this point? But it's not entirely clear for me if this is actually 100% guaranteed.

Reply:
Any reason we are not using `this.orgHintCookieName` here instead of `buildOrganizationHintCookieName(this.options.client_id)` ?

However, curious to know if it would make sense to drop `this.orgHintCookieName` and use `buildOrganizationHintCookieName(this.options.client_id)` everywhere. It would allow to work if `client_id` is ever changed after constructing an Auth0Client (Not sure we should explicitly support that, but we could).

Reply:
> Why are we not setting the cookie here?

We could set it here as well, I think that's fine.

> Any reason we are not using this.orgHintCookieName here

I think that's just an oversight, I'd like to use it here as well. I did start down the road of just calling the function but I think if we wanted to support changing the client ID (can't think of a use-case right now) there'd be more work to do than just this.

Reply:
I will also need to review `getTokenWithPopup` as well

Reply:
Fixed in https://github.com/auth0/auth0-spa-js/pull/788/commits/0211ca7f19bd90626ae6539a21ae49317621cfba. `getTokenWithPopup` looks to be ok as it just calls `loginWithPopup` anyway.